elements/pfe-clipboard/src/pfe-clipboard.js

```diff
@@ -0,0 +1,254 @@
+   * // attribute where the destination slot name is provided.
+   * // <slot hidden data-slot="text"></slot>
+   * _slotchangeHandler(event) { this._transposeSlot(event) }
+   *
+   * @param {event} SlotchangeEvent
+   * @return void
+   */
+  _transposeSlot(event) {
+    const source = event.target;
+    const slot = source.dataset.slot;
+    const target = this.shadowRoot.querySelector(`slot[name="${slot}"]`);
+    const fallbackContentVariable = `_transposeSlot${slot}`;
+    // Store the default content of the target for later use.
+    if (typeof this[fallbackContentVariable] === "undefined") {
+      this[fallbackContentVariable] = target.innerHTML;
+    }
+    // Get all children for the source slot. We need to include flatten to make sure we get the content.
+    const childNodes = [...source.assignedNodes({ flatten: true })];
+    // Trim the content nodes for whitespace and combine it into one string for evaluation.
+    const childContent = childNodes.map(child => child.textContent.trim()).join("");
+    // Find out if there is any non whitespace content
+    if (childContent !== "") {
+      // Transpose the content to the target slot
+      target.innerHTML = childContent;
+    }
+    // If there isn't any content then we are going to place the target slot's default content back in.
+    else {
+      target.innerHTML = this[fallbackContentVariable];
+    }
+  }
+
+  disconnectedCallback() {
+    this.removeEventListener("click", this._clickHandler.bind(this));
+    this.removeEventListener("keydown", this._keydownHandler.bind(this));
+    this.shadowRoot.removeEventListener("slotchange", this._slotchangeHandler.bind(this));
+    super.disconnectedCallback();
+  }
+
+  _noIconChanged(previousValue) {
+    // dirty check to see if we should rerender the template
+    if (this._rendered && this.noIcon !== previousValue) {
+      this.render();
+    }
+  }
+
+  _clickHandler(event) {
+    // Execute the copy to clipboard functionality
+    this.copyURLToClipboard()
+      .then(url => {
+        // Emit event that lets others know the user has "copied"
+        // the url. We are also going to include the url that was
+        // copied.
+        this.emitEvent(PfeClipboard.events.copied, {
+          detail: {
+            url
+          }
+        });
+        // Toggle the copied state. Use the this._formattedCopiedTimeout function
+        // to an appropraite setTimout length.
+        this.setAttribute("copied", "");
+        setTimeout(() => {
+          this.removeAttribute("copied");
+        }, this._formattedCopiedTimeout());
+      })
+      .catch(error => {
+        this.warn(error);
+      });
+  }
+
+  // Formatted copied timeout value. Use the formattedCopiedTimeout function
+  // to get a type safe, millisecond value of the timeout duration.
+  _formattedCopiedTimeout() {
+    const copiedDuration = Number(this.copiedDuration * 1000);
+    if (!Number.isInteger(copiedDuration)) {
```

Comment:
You don't need to type-check the copied duration attribute because it is defined as type Number, PFElement will type-check it automatically.  I would keep the default but remove the warning message.